### DISCUSSION TITLE
Asking for guidance to understand the project [3]Multi-channel Imaging Support.

### DISCUSSION THREAD

**Discussion by kamlesh-ops** [camicroscope/GSOC#45] [General]
Hello sir,
I am Kamlesh Sahoo, an undergraduate B.Tech student at the Indian Institute of Technology, Dhanbad, India. I know C++, python, ML frameworks like Tensorflow and Pytorch, OpenCV, and Image processing, alongside different ML algorithms and libraries like Numpy, pandas, seaborn, and sci-kit-learn.
I am here with a desire to make open-source contributions and be a part of GSOC 2024. 
In the list of projects for 2023, I found this project [3]Multi-channel Imaging Support, and I seek some guidance to understand this project. 
@nanli-emory it would be really helpful for me sir if u could guide me on how to go about the project. It would also be great if u could share some related resources(like source code) for a similar project.
Looking forward to your response.
Thank you :)

**Comment by birm** [comment]
Hi and thank you for reaching out! I'm not sure we'll have this as a GSOC project for next year. We haven't gotten around to updating the list yet. I'll add an issue to this repo to mark that we need to update the list!

If you're interested in contributing anyway, we have noticed some images have more channels than RGB. This unrelated medium post might show some of the problem: https://medium.com/abraia/hyperspectral-image-analysis-with-python-made-easy-68dcd0234924. I think a good implementation would be a special service which acts as a tile service for specific channels, and an integration to caMicroscope that lets users mix them as they see fit.

**Reply by kamlesh-ops** [reply]
Thank you sir for responding!
I will work on your suggestions and get back if I can get something. Also, I'll look forward to the updated list.  
Thanks :)

**Reply by kamlesh-ops** [reply]
Hello @birm @nanli-emory I was just going through the updated list for GSOC 2024.I find that the previous project [3] Multi Channel Imaging support has some correlation with the project High-Dimensional Imaging support, as you had said this won't be a part of  this year. Is that so?
Also, I was going through the above link you shared on hyperspectral images. You talked about the tile service. So basically it is needed that the channels should be read one at a time dynamically right?
Please let me know how should I start and proceed to work on the above-mentioned issue. I also believe that this will help me to understand the new project on High-Dimensional Imaging Support better. 
Thanks:)

**Reply by birm** [reply]
We reworked the project a little bit from last year, and thought that high dimensional imaging was a more accurate name.
You can propose something different in the project proposal, but the one channel at a time seems to be a reasonable/flexible approach to me. Though it may have http overhead impacting performance, it should still _hopefully_ be reasonable, and I don't think we have to worry about optimizing performance so long as it's responsive to the user.

Hope this helps!

**Reply by kamlesh-ops** [reply]
@birm Thanks for the response :)
Okay got it. One small question, along with the proposal, would it be better if I provide small snippets suggesting implementation of the problem. Or anything else that can make my approach more clear, then please suggest me.
